lib/segment/src/common/operation_time_statistics.rs

```diff
@@ -87,9 +138,39 @@ impl std::ops::Add for OperationDurationStatistics {
                 other.max_duration_micros,
                 |a, b| a > b,
             ),
+            total_duration_micros: self.total_duration_micros + other.total_duration_micros,
             last_responded: std::cmp::max(self.last_responded, other.last_responded),
+            duration_micros_histogram: merge_buckets(
+                self.duration_micros_histogram,
+                other.duration_micros_histogram,
+            ),
+        }
+    }
+}
+
+/// Merges two vectors of buckets, summing the counts of the same boundaries.
+/// # Panics
+/// Panics when merging buckets of different configurations.  In practice, we should stick to a
+/// single configuration so this should not happen.
+fn merge_buckets(mut a: Vec<(f32, usize)>, b: Vec<(f32, usize)>) -> Vec<(f32, usize)> {
```

Comment:
This may be slightly better:

```suggestion
fn merge_buckets(mut a: Vec<(f32, usize)>, b: &[(f32, usize)]) -> Vec<(f32, usize)> {
```

Reply:
Both `a` and `b` could be `return`ed, so both of them has to be owning types.

Reply:
You're totally right.